app/models/merchant.rb

```diff
@@ -27,16 +27,37 @@ def items_ready_to_ship
       .order(:created_at)
   end
 
-  def top_items_by_revenue
+  def top_items_by_revenue #top 5 most popular items
     items
       .joins(invoices: :transactions)
-      .where('transactions.result = 0')
+      .where('transactions.result = 0 AND invoices.status = 1')
       .select('items.*, SUM(invoice_items.quantity * invoice_items.unit_price) AS item_revenue')
       .group(:id)
       .order(item_revenue: :desc)
       .limit(5)
   end
 
+  def most_popular_items
+    items
+      .joins(invoice_items: :invoice)
+      .where('invoices.status = 1')
+      .select('items.*, SUM(invoice_items.quantity) AS total_sold')
+      .group(:id)
+      .order(total_sold: :desc)
+      .limit(5)
+  end
+
+  def top_selling_date
```

Comment:
I believe this has a model test, but we still need to implement the view page and corresponding tests